File: exercise07-A/src/program.py
#!/usr/bin/env python3

# Exercise 07-A: Cookies

import sys

# Functions


def readline() -> list[int]:
    # Read a line from stdin
    return [int(i) for i in sys.stdin.readline().split()]


def feed_children(children: list[int], cookies: list[int]) -> int:
    # Return number of children fed with cookies
    children.sort(reverse=True)
    cookies.sort()

    content_children = 0
    for child in children:
        if cookies[-1] >= child:
            cookies.pop()
            content_children += 1

    return content_children


# Main execution


def main() -> None:
    while (children := readline()) and (cookies := readline()):
        print(feed_children(children, cookies))


if __name__ == "__main__":
    main()

File: exercise07-A/src/program.cpp
// Exercise 07-A: Cookies

#include <algorithm>
#include <deque>
#include <iostream>
#include <sstream>
#include <string>

// Functions

bool readline(std::deque<int> &v) {
    // TODO: Read a line from stdin
    return false;
}

int feed_children(std::deque<int> &children, std::deque<int> &cookies) {
    // TODO: Return number of children fed with cookies
    return 0;
}

// Main execution

int main(int argc, char *argv[]) {
    std::deque<int> children;
    std::deque<int> cookies;

    while (readline(children) && readline(cookies)) {
        std::cout << feed_children(children, cookies) << std::endl;
    }

    return 0;
}
